DPGAnalysis/Skims/src/ZElectronsSelector.cc

```diff
@@ -0,0 +1,197 @@
+}
+
+
+
+void  ZElectronsSelector::newEvent(const edm::Event& ev, const edm::EventSetup& ){
+
+  ev.getByToken(theRhoToken,_rhoHandle);
+  
+}
+
+bool ZElectronsSelector::operator()(const reco::GsfElectron& el) const{
+
+  bool keepEle = false; 
+  float pt_e = el.pt();
+
+  auto etrack  = el.gsfTrack().get(); 
+
+  if (el.isEB()){
+    if (etrack->hitPattern().numberOfLostHits(reco::HitPattern::MISSING_INNER_HITS)>missHits[0]) return keepEle; 
+    if (el.full5x5_sigmaIetaIeta() >  sigmaIEtaIEtaCut[0]) return keepEle; 
+    if (fabs(el.deltaPhiSuperClusterTrackAtVtx())> dPhiInCut[0]) return keepEle; 
+    if (fabs(el.deltaEtaSeedClusterTrackAtVtx())> dEtaInSeedCut[0]) return keepEle; 
+    if (el.hadronicOverEm()>hOverECut[0]) return keepEle;
+    float abseta = fabs((el.superCluster().get())->position().eta());
+    if (abseta> 1.479) return keepEle; // check if it is really needed
+    const float  eA = getEffectiveArea( abseta );
+    const float rho = _rhoHandle.isValid() ? (float)(*_rhoHandle.product()) : 0; 
+    if (( el.pfIsolationVariables().sumChargedHadronPt + 
+	  std::max(float(0.0), el.pfIsolationVariables().sumNeutralHadronEt +  
+		   el.pfIsolationVariables().sumPhotonEt -  eA*rho )
+	  ) > relCombIso[0]*pt_e) return keepEle; 
+    const float ecal_energy_inverse = 1.0/el.ecalEnergy();
+    const float eSCoverP = el.eSuperClusterOverP();
+    if ( std::abs(1.0 - eSCoverP)*ecal_energy_inverse > EInvMinusPInv[0]) return keepEle; 
+    keepEle=true;
+    
+  }else if (el.isEE()){
+    if (etrack->hitPattern().numberOfLostHits(reco::HitPattern::MISSING_INNER_HITS)>missHits[1]) return keepEle; 
+    if (el.full5x5_sigmaIetaIeta() >  sigmaIEtaIEtaCut[1]) return keepEle; 
+    if (fabs(el.deltaPhiSuperClusterTrackAtVtx())> dPhiInCut[1]) return keepEle; 
+    if (fabs(el.deltaEtaSeedClusterTrackAtVtx())> dEtaInSeedCut[1]) return keepEle; 
+    if (el.hadronicOverEm()>hOverECut[1]) return keepEle;
+    float abseta = fabs((el.superCluster())->position().eta());
+    if (abseta< 1.479) return keepEle; // check if it is really needed
+    if (abseta>=2.5  ) return keepEle; // check if it is really needed
+    
+    const float  eA = getEffectiveArea( abseta );
+    const float rho = _rhoHandle.isValid() ? (float)(*_rhoHandle) : 0; 
+    if ((el.pfIsolationVariables().sumChargedHadronPt + 
+	 std::max(float(0.0), el.pfIsolationVariables().sumNeutralHadronEt + 
+		 el.pfIsolationVariables().sumPhotonEt - eA*rho)
+	 ) >relCombIso[1]*pt_e) return keepEle; 
+    const float ecal_energy_inverse = 1.0/el.ecalEnergy();
+    const float eSCoverP = el.eSuperClusterOverP();
+    if ( std::abs(1.0 - eSCoverP)*ecal_energy_inverse > EInvMinusPInv[1]) return keepEle; 
+
+    keepEle=true;
+	
+  }//isEE
+  return keepEle;
+
+}
+
+EVENTSETUP_STD_INIT(ZElectronsSelector);
+#include "CommonTools/UtilAlgos/interface/SingleObjectSelector.h"
+#include "CommonTools/UtilAlgos/interface/AndSelector.h"
```

Comment:
why are this and the following two includes needed?

Reply:
No, removed